spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkDistributionAndOrderingUtil.java

```diff
@@ -71,7 +72,8 @@ public static Distribution buildRequiredDistribution(Table table, DistributionMo
   }
 
   public static SortOrder[] convert(org.apache.iceberg.SortOrder sortOrder) {
-    List<OrderField> converted = SortOrderVisitor.visit(sortOrder, new SortOrderToSpark());
+    Map<Integer, String> quotedNameById = SparkSchemaUtil.indexQuotedNameById(sortOrder.schema());
```

Comment:
I am a bit worried about the performance. We index all columns. We could add some sort of a filter to the indexing logic to only pick particular field IDs. It will be more involved so let me know what everybody thinks.

Reply:
I don't think this is going to matter much. Schema operations are done in planning, not in a tight loop. And even with thousands of columns, this shouldn't take that long. If it were done in a loop, I think we'd want to care.

Reply:
I'll still probably address it in a follow-up PR just to be safe.